docs/add-runtime.md

```diff
@@ -0,0 +1,250 @@
+
+### 1.1 Fork the OpenRuntimes repository
+
+Before making any changes, you will need to fork OpenRuntimes's repository to keep branches on the official repo clean. To do that, visit [OpenRuntimes's Runtime repository](https://github.com/open-runtimes/open-runtimes) and click on the fork button.
+
+[![Fork button](https://github.com/appwrite/appwrite/raw/master/docs/tutorials/images/fork.png)](https://github.com/appwrite/appwrite/blob/master/docs/tutorials/images/fork.png)
+
+This will redirect you from `github.com/open-runtimes/open-runtimes` to `github.com/YOUR_USERNAME/open-runtimes`, meaning all changes you do are only done inside your repository. Once you are there, click the highlighted `Code` button, copy the URL and clone the repository to your computer using the `git clone` command:
+```bash
+$ git clone COPIED_URL
+```
+
+> To fork a repository, you will need a basic understanding of CLI and git-cli binaries installed. If you are a beginner, we recommend you to use `Github Desktop`. It is a clean and simple visual Git client.
+
+Finally, you will need to create a `feat-XXX-YYY-runtime` branch based on the `refactor` branch and switch to it. The `XXX` should represent the issue ID and `YYY` the runtime name.
+
+## 2. Implement new runtime
+
+### 2.1 Preparing the files for your new runtime
+The first step to writing a new runtime is to create a folder within `/runtimes` with the name of the runtime and the version separated by a dash. For instance, if I was to write a Rust Runtime with version 1.55 the folder name would be: `rust-1.55`
+
+Within that folder you will need to create a few basic files that all OpenRuntimes runtimes require:
+```
+Dockerfile - Dockerfile that explains how the container will be built.
+README.md - A readme file explaining the runtime and any special notes for the runtime. A good example of this is the PHP 8.0 runtime.
+```
+
+### 2.2 Differences between compiled and interpreted runtimes
+Runtimes within OpenRuntimes are created differently depending on whether they are compiled or interpreted. This is due to the fundamental differences between the two ways of running the code.
+
+Interpreted languages have both a `build.sh` file and a `start.sh` file.
+The `build.sh` file for an interpreted runtime is normally used for installing any dependencies for both the server itself and the user's code and then to copy it to the `/usr/code` folder which is then packaged and can be used later for running the server.
+The build script is always executed during the build stage of a function deployment.
+
+The `start.sh` file for an interpreted runtime should extract the `/tmp/code.tar.gz` file that contains both the user's code and the dependencies. This tarball was created by OpenRuntimes from the `/usr/code` folder and should install the dependencies that were pre-installed by the build stage and move them into the relevant locations for that runtime. It will then run the server ready for execution.
```

Comment:
```suggestion
The `start.sh` file for an interpreted runtime should extract the `/tmp/code.tar.gz` file that contains both the user's code and the dependencies. This tarball was created by Open Runtimes from the `/usr/code` folder and should install the dependencies that were pre-installed by the build stage and move them into the relevant locations for that runtime. It will then run the server ready for execution.
```